gcloud-java-examples/src/main/java/com/google/cloud/examples/nio/ParallelCountBytes.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.examples.nio;
+
+import com.google.common.base.Stopwatch;
+
+import javax.xml.bind.annotation.adapters.HexBinaryAdapter;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.ByteBuffer;
+import java.nio.channels.SeekableByteChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.MessageDigest;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * ParallelCountBytes will read through the whole file given as input.
+ *
+ * <p>This example shows how to go through all the contents of a file,
+ * in order, using multithreaded NIO reads.It also reports how long it took.
+ *
+ * <p>See the README for compilation instructions. Run this code with
+ * {@code target/appassembler/bin/ParallelCountBytes <file>}
+ */
+public class ParallelCountBytes {
+
+  private class BufWithLock {
+    public Object lock;
+    public ByteBuffer buf;
+    public boolean full;
+    public Thread t;
```

Comment:
Not really, no. Sometimes it's handy to keep the Thread object around to be able to interrupt it, but we're not doing that here. I removed it.